### ISSUE TITLE
Support more platforms with ENABLE_MASON=ON

### ISSUE THREAD

**Issue by springmeyer** [Project-OSRM/osrm-backend#3480]
OS X >= 10.10 and Ubuntu >= 12.04 are the two main platforms that ENABLE_MASON=ON builds work well on and are well tested ([per wiki documentation](https://github.com/Project-OSRM/osrm-backend/issues/3207#issuecomment-268555489)).

This ticket is to track adding support for more platforms. This is low urgency, but nice to have.

 - [ ] Systems with libstdc++ older than 5.x and no ability to upgrade will hit https://github.com/Project-OSRM/osrm-backend/issues/3207#issuecomment-258020247. (Will be unblocked by https://github.com/mapbox/mason/issues/267)
 - [ ] Arch linux only works currently by [turning off LTO and Gold linking](https://github.com/Project-OSRM/osrm-backend/pull/3119#issuecomment-256790635). The task is to figure out a workaround to enable LTO and ENABLE_MASON on arch.

**Comment by springmeyer** [comment]
> This is low urgency, but nice to have.

Therefore I'm considering this ticket as not actionable and not valuable open. Closing since there is no active work planned to address ancient libstdc++ or time spent to debug arch linux binutils+clang+lto issues (which might be solvable but don't have an obvious solution).
